core/src/main/java/io/questdb/griffin/engine/functions/SymbolFunction.java

```diff
@@ -165,5 +165,21 @@ public StaticSymbolTable getStaticSymbolTable() {
         return null;
     }
 
+    /**
+     * A clone of function's symbol table to enable concurrent SQL execution.
+     * During such execution symbol table clones will be assigned to individual executing
+     * thread.
+     * @return clone of symbol table
+     */
+    @Nullable
+    public SymbolTable newSymbolTable() {
+        return null;
+    }
+
     public abstract boolean isSymbolTableStatic();
+
+    @Override
+    public boolean isReadThreadSafe() {
+        return false;
```

Comment:
consider `isConstant()` All constants should be thread-safe. I would think. 
this impl makes e.g. the `StrConstant` return `false`

Reply:
it is already the case via inheritance:

https://github.com/questdb/questdb/blob/vi_fix_async_symbol/core/src/main/java/io/questdb/griffin/engine/functions/constants/ConstantFunction.java

Reply:
the default impl in interface works unless there is another impl in the inheritance hierarchy. consider this:
`StrConstant implements ConstantFunction` which has:
```
default boolean isReadThreadSafe() {
    return true;
}
```
but it also inherits from `StrFunction` which has:
```
    public boolean isReadThreadSafe() {
        return false;
    }
```
the function from StrFunction takes a precedence over the default impl from the interface -> `StrConstant` returns `false` even it's `ConstatnFunction`. 
Changing `StrFunction` to:
```
   public boolean isReadThreadSafe() {
        return isConstant();
    }
```
would help. IT's not a blocker, but rather another illustration how fiddly this is.